addons/html_editor/static/src/core/dom_plugin.js

```diff
@@ -643,9 +654,14 @@ export class DomPlugin extends Plugin {
                 newCandidate.append(...childNodes(block));
                 block.append(newCandidate);
                 cursors.remapNode(block, newCandidate);
+                newCandidate = createNewCandidate();
             }
         }
         cursors.restore();
+        // Split the block segments again in case some line breaks remained
+        // because of unsplittable blocks that were now changed to contain
+        // splittable ones.
+        this.dependencies.split.splitBlockSegments();
```

Comment:
This is probably legit, but your comment did not helped me understand the use case.
So it probably would be nice to rewrite the comment or provide better context in some way...

🤷‍♂️

Reply:
Good catch, actually with the recent change to handle unsplittable blocks (inserting splittable blocks in them instead of doing nothing), this was not needed anymore.